client/renderSDL/RenderHandler.cpp

```diff
@@ -55,6 +55,58 @@ std::shared_ptr<CDefFile> RenderHandler::getAnimationFile(const AnimationPath &
 	return result;
 }
 
+std::optional<ResourcePath> RenderHandler::getPath(ResourcePath path)
+{
+	if(CResourceHandler::get()->existsResource(path))
+		return path;
+	if(path.getType() == EResType::IMAGE)
+	{
+		auto p = ImagePath::builtin(path.getName());
+		if(CResourceHandler::get()->existsResource(p.addPrefix("DATA/")))
+			return std::optional<ResourcePath>(p.addPrefix("DATA/"));
+		if(CResourceHandler::get()->existsResource(p.addPrefix("SPRITES/")))
+			return std::optional<ResourcePath>(p.addPrefix("SPRITES/"));
+	}
+	else
+	{
+		auto p = AnimationPath::builtin(path.getName());
+		auto pJson = p.toType<EResType::JSON>();
+		if(CResourceHandler::get()->existsResource(p.addPrefix("SPRITES/")))
+			return std::optional<ResourcePath>(p.addPrefix("SPRITES/"));
+		if(CResourceHandler::get()->existsResource(pJson))
+			return std::optional<ResourcePath>(p);
+		if(CResourceHandler::get()->existsResource(pJson.addPrefix("SPRITES/")))
+			return std::optional<ResourcePath>(p.addPrefix("SPRITES/"));
+	}
+
+	return std::nullopt;
+}
+
+std::pair<ResourcePath, int> RenderHandler::getScalePath(ResourcePath p)
+{
+	auto path = p;
+	auto name = p.getName();
+	int scaleFactor = 1;
+	if(getScalingFactor() > 1)
+	{
+		std::vector<int> factorsToCheck = {getScalingFactor(), 4, 3, 2};
```

Comment:
Why we need to check multiple scaling factors? We should only try to load one that matches desired one.

Reply:
Because also a downscaling is applied, example:
We only have x4 images. User selected x2. Then the images are downscaled to x2.
If there are also x2 images they are used without scaling.